winstore/Program.cs

```diff
@@ -0,0 +1,271 @@
+﻿using CsvHelper;
+using System.Globalization;
+using System.IO.Compression;
+using System.Text;
+
+namespace convert_to_winstore
+{
+    /// <summary>
+    /// Class representing an OpenTTD language.
+    /// </summary>
+    internal class Language
+    {
+        /// <summary>
+        /// Language filename base (e.g. "english");
+        /// </summary>
+        public string LangFile { get; }
+
+        /// <summary>
+        /// Windows Store language code (e.g. "en-gb");
+        /// </summary>
+        public string StoreCode { get; }
+
+        /// <summary>
+        /// Loaded strings for substitution.
+        /// </summary>
+        private Dictionary<string, string> strings;
+
+        /// <summary>
+        /// Creates a new instance of the Language class.
+        /// </summary>
+        /// <param name="langFile">Language filename base.</param>
+        /// <param name="storeCode">Windows Store language code.</param>
+        public Language(string langFile, string storeCode)
+        {
+            LangFile = langFile;
+            StoreCode = storeCode;
+        }
+
+        /// <summary>
+        /// Replaces $VARIABLES$ within the specified string with translated strings.
+        /// </summary>
+        /// <param name="langPath">Path to the 'lang' folder.</param>
+        /// <param name="stringToSubstitute">String to work on.</param>
+        /// <returns>The substituted string.</returns>
+        public string Substitute(string langPath, string stringToSubstitute)
+        {
+            if (strings == null)
+            {
+                strings = new Dictionary<string, string>(StringComparer.Ordinal);
+                string path = Path.Combine(langPath, Path.ChangeExtension(LangFile, ".txt"));
+
+                if (!File.Exists(path))
+                {
+                    // TODO: For now, substitute the English version.
```

Comment:
The downside of `TODO`s: who is going to pick it up when?

So maybe the question .. is it actually a TODO?

Reply:
Well, the hope is that once the translations are in EINTS, people translate them, and we no longer need this. :) So I suppose it depends on how long the translations take.

Reply:
Wouldn't it always be better to use the english sentence for those translations that do not have a translated string? That is more why I meant if this is actually a TODO. As I don't think you want to do it "for now", but in all cases where the translation is without an actual translation :)

Reply:
Owh, I was misreading the if-statement .. now you created all the language files, is this code that is actually executed?